--- origsrc/gnash-0.8.9/extensions/dbus/Makefile.am	2011-03-12 10:52:05.000000000 -0600
+++ src/gnash-0.8.9/extensions/dbus/Makefile.am	2011-04-08 03:05:45.934119900 -0500
@@ -35,7 +35,7 @@ AM_CPPFLAGS =  \
 dbus_la_SOURCES = dbus_ext.cpp dbus_ext.h
 dbus_la_CPPFLAGS = $(AM_CPPFLAGS)
 dbus_la_LDFLAGS = -module -avoid-version -no-undefined
-dbus_la_LIBDADD = $(LIBADD_DL) \
+dbus_la_LIBADD = $(LIBADD_DL) \
 	$(top_builddir)/libbase/libgnashbase.la \
 	$(top_builddir)/libcore/libgnashcore.la \
 	$(BOOST_LIBS) \
@@ -49,7 +49,7 @@ CLEANFILES = \
 install-pluginsLTLIBRARIES: $(plugins_LTLIBRARIES)
 	test -d "$(DESTDIR)$(pluginsdir)" || $(mkinstalldirs) "$(DESTDIR)$(pluginsdir)"
 	$(LIBTOOL) --mode=install $(INSTALL) $(INSTALL_STRIP_FLAG) $(plugins_LTLIBRARIES) "$(DESTDIR)$(pluginsdir)/$(plugins_LTLIBRARIES)"
-	$(RM) $(DESTDIR)$(pluginsdir)/*.a 
+	$(RM) $(DESTDIR)$(pluginsdir)/*.*a 
 
 uninstall-local:
 	$(RM) $(DESTDIR)$(pluginsdir)/dbus.so
--- origsrc/gnash-0.8.9/extensions/fileio/Makefile.am	2011-03-12 10:52:05.000000000 -0600
+++ src/gnash-0.8.9/extensions/fileio/Makefile.am	2011-04-08 01:42:29.520795100 -0500
@@ -48,7 +48,9 @@ AM_LDFLAGS =  \
 
 fileio_la_SOURCES = fileio.cpp fileio.h
 fileio_la_LDFLAGS = -module -avoid-version -no-undefined
-fileio_la_LIBDADD = $(LIBADD_DL) $(AM_LDFLAGS)
+fileio_la_LIBADD = $(LIBADD_DL) $(AM_LDFLAGS) \
+	$(top_builddir)/libbase/libgnashbase.la \
+	$(top_builddir)/libcore/libgnashcore.la
 
 
 check_SCRIPTS =
--- origsrc/gnash-0.8.9/extensions/mysql/Makefile.am	2011-03-12 10:52:05.000000000 -0600
+++ src/gnash-0.8.9/extensions/mysql/Makefile.am	2011-04-08 02:11:40.342649200 -0500
@@ -43,7 +43,9 @@ AM_LDFLAGS =  \
 
 mysql_la_SOURCES = mysql_db.cpp mysql_db.h
 mysql_la_LDFLAGS = -module -avoid-version -no-undefined
-mysql_la_LIBDADD = $(AM_LDFLAGS)
+mysql_la_LIBADD = $(AM_LDFLAGS) \
+	$(top_builddir)/libbase/libgnashbase.la \
+	$(top_builddir)/libcore/libgnashcore.la
 
 CLEANFILES = \
       gnash-dbg.log
--- origsrc/gnash-0.8.8/libbase/Makefile.am	2010-08-07 10:50:00.000000000 -0500
+++ src/gnash-0.8.8/libbase/Makefile.am	2010-09-28 01:31:22.028404600 -0500
@@ -55,7 +55,7 @@ libgnashbase_la_LIBADD = \
 	$(Z_LIBS) \
 	$(CURL_LIBS) \
 	$(OPENGL_LIBS) \
-	$(LIBINTL) \
+	$(LTLIBINTL) \
 	$(BOOST_LIBS) \
 	$(PTHREAD_LIBS) \
 	$(NULL)
--- origsrc/gnash-0.8.9/librender/Makefile.am	2011-03-13 11:47:36.000000000 -0500
+++ src/gnash-0.8.9/librender/Makefile.am	2011-04-08 00:17:07.012283500 -0500
@@ -93,6 +93,10 @@ libgnashrender_la_SOURCES += cairo/PathP
 libgnashrender_la_LIBADD += $(CAIRO_LIBS)
 endif
 
+libgnashrender_la_LIBADD += \
+	$(top_builddir)/libbase/libgnashbase.la \
+	$(top_builddir)/libcore/libgnashcore.la
+
 # Rebuild with GCC 4.x Mudflap support
 mudflap:
 	@echo "Rebuilding with GCC Mudflap support"
--- origsrc/gnash-0.8.8/plugin/klash4/Makefile.am	2010-08-07 10:50:01.000000000 -0500
+++ src/gnash-0.8.8/plugin/klash4/Makefile.am	2010-09-28 01:33:00.166193700 -0500
@@ -40,7 +40,7 @@ libklashpart_la_DEPENDENCIES = klash_par
 if PLUGIN_LINK_UNDEFINED
 libklashpart_la_LDFLAGS = -avoid-version -L$(plugindir) $(KDE4_LIBS) $(QT4_LIBS)
 else
-libklashpart_la_LDFLAGS = -module -avoid-version -L$(plugindir) $(KDE4_LIBS) $(QT4_LIBS)
+libklashpart_la_LDFLAGS = -module -avoid-version -no-undefined -L$(plugindir) $(KDE4_LIBS) $(QT4_LIBS)
 endif
 
 dist_kde_services_DATA = klash_part.desktop
--- origsrc/gnash-0.8.8/plugin/npapi/Makefile.am	2010-08-08 20:54:32.000000000 -0500
+++ src/gnash-0.8.8/plugin/npapi/Makefile.am	2010-10-04 23:18:48.160017900 -0500
@@ -94,7 +94,7 @@ check:
 if PLUGIN_LINK_UNDEFINED
 libgnashplugin_la_LDFLAGS = -avoid-version -L$(plugindir)
 else
-libgnashplugin_la_LDFLAGS = -module -avoid-version -L$(plugindir)
+libgnashplugin_la_LDFLAGS = -module -avoid-version -no-undefined -L$(plugindir)
 endif
 
 if VISIBILITY_WORKS
@@ -120,11 +120,7 @@ if !WIN32
 install-plugin install-plugins: $(plugin_LTLIBRARIES) $(PKGLIB)
 	@test -d "$(DESTDIR)$(plugindir)" || $(mkinstalldirs) "$(DESTDIR)$(plugindir)"
 	@echo "Installing plugin to $(DESTDIR)$(plugindir)"
-	@if test -d .libs; then \
-	  cp -f .libs/libgnashplugin.so* $(DESTDIR)$(plugindir)/libgnashplugin.so; \
-	else \
-	  cp -f libgnashplugin.so* $(DESTDIR)$(plugindir)/libgnashplugin.so; \
-	fi
+	$(LIBTOOL) --mode=install $(INSTALL) $(INSTALL_STRIP_FLAG) $(plugin_LTLIBRARIES) $(DESTDIR)$(plugindir)
 else
 install-plugin install-plugins: 
 	: # do nothing
