--- origsrc/gnash-0.8.8/gui/Makefile.am	2010-08-07 10:50:00.000000000 -0500
+++ src/gnash-0.8.8/gui/Makefile.am	2010-09-28 01:29:22.930679800 -0500
@@ -55,7 +55,7 @@ AM_CPPFLAGS = \
         -I$(top_srcdir)/libcore/parser \
         -I$(top_srcdir)/libcore/vm \
         -I$(top_srcdir)/libcore/asobj \
-	-I$(top_srcdir)/libltdl \
+	$(INCLTDL) \
         -I$(top_srcdir)/libbase \
         -I$(top_srcdir)/librender \
 	-I$(top_srcdir)/libmedia \
@@ -121,7 +121,7 @@ GNASH_LIBS = \
 	$(top_builddir)/libbase/libgnashbase.la
 
 if LIBLTDL2
-GNASH_LIBS += $(top_builddir)/libltdl/libltdlc.la
+GNASH_LIBS += $(LIBLTDL)
 endif
 if LIBLTDL1
 GNASH_LIBS += $(top_builddir)/libbase/libltdlc.la
--- origsrc/gnash-0.8.8/libbase/Makefile.am	2010-08-07 10:50:00.000000000 -0500
+++ src/gnash-0.8.8/libbase/Makefile.am	2010-09-28 01:31:22.028404600 -0500
@@ -29,7 +29,7 @@ pkglib_LTLIBRARIES = libgnashbase.la
 
 libgnashbase_la_CPPFLAGS = -DPLUGINSDIR=\"$(pluginsdir)\" \
 	-DSYSCONFDIR=\"$(sysconfdir)\" \
-	-I$(top_srcdir)/libltdl \
+	$(INCLTDL) \
 	$(PTHREAD_CFLAGS) \
 	$(PNG_CFLAGS) \
 	$(GIF_CFLAGS) \
@@ -130,7 +130,7 @@ libgnashbase_la_LIBADD += $(top_builddir
 endif
 
 if LIBLTDL2
-libgnashbase_la_LIBADD += $(top_builddir)/libltdl/libltdlc.la
+libgnashbase_la_LIBADD += $(LIBLTDL)
 endif
 
 
