--- origsrc/gnash-0.8.2/configure.ac	2008-03-04 09:53:27.000000000 -0600
+++ src/gnash-0.8.2/configure.ac	2008-05-11 16:10:55.984375000 -0500
@@ -739,7 +739,6 @@
 AM_CONDITIONAL(BUILD_METOME_EXT, [ test x$ext_metome = xyes ])
 AM_CONDITIONAL(BUILD_EXTENSIONS, [ test -n "$extensions_list"])
 
-AC_LIBLTDL_INSTALLABLE
 AC_LIBTOOL_DLOPEN
 AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
@@ -1260,8 +1259,12 @@
 AC_CHECK_FUNCS(strlcpy, AC_DEFINE(HAVE_STRLCPY_PROTO, [1],[Define if you have the strlcpy prototype]))
 AC_CHECK_FUNCS(strlcat, AC_DEFINE(HAVE_STRLCAT_PROTO, [1],[Define if you have the strlcat prototype]))
 dnl Look for Win32 networking stuff
+case $host_os in
+mingw*)
 AC_CHECK_HEADERS(winsock.h)
 AC_CHECK_HEADERS(winsock2.h)
+;;
+esac
 AC_CHECK_FUNCS(socket)
 AC_CHECK_FUNCS(CreateFileMappingA)
 
@@ -1670,9 +1673,11 @@
 
 dnl Define winsock if we're on windows. We could do something complicated,
 dnl but since AC_EXEEXT does it for us, we'll do this the easy way.
-if test x"$EXEEXT" = "exe"; then
+case $host_os in
+mingw*)
   AC_DEFINE(HAVE_WINSOCK,1,[This is defined is we are on Win32])
-fi
+;;
+esac
 
 dnl ****************************************
 dnl *** Check for ELF visibility support ***
--- origsrc/gnash-0.8.2/libbase/utility.h	2008-01-21 14:55:46.000000000 -0600
+++ src/gnash-0.8.2/libbase/utility.h	2008-05-11 16:01:05.453125000 -0500
@@ -120,7 +120,7 @@
 //This is from C99.
 const float LN_2 = 0.693147180559945f;
 // the overridden log(f) will use logf IFF f is a float
-#ifndef HAVE_LOG2
+#if !defined HAVE_LOG2 && !defined log2
 inline double	log2(double f) { return std::log(f) / LN_2; }
 #endif
 //exp2 might be missing on Net-/OpenBSD.
